pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Environment.groovy

```diff
@@ -94,19 +97,69 @@ public class Environment implements Serializable {
         unsetKeys.addAll(overrides.keySet())
 
         GroovyShell shell = new GroovyShell(new Binding(alreadySet))
-        shell.setProperty("params", (Map<String,String>)script.getProperty("params"))
+        shell.setProperty("params", (Map<String, String>) script.getProperty("params"))
+
+        Map<String, String> preCreds = roundRobin(shell, alreadySet, overrides, unsetKeys)
+
+        Run<?, ?> r = script.$build()
```

Comment:
It's getting the build - simpler then using `currentBuild` in a non script context.